### DISCUSSION TITLE
Deprecated classes and now what...

### DISCUSSION THREAD

**Discussion by JorgeBarto** [eclipse-edc/Connector#4404] [General]
Hi everyone,
just a 'simple' question; when I see a **deprecated** class such as org.eclipse.edc.connector.controlplane.transfer.dataplane.flow.**ConsumerPullTransferDataFlowController**, which has been deprecated since 0.5.1 version, where should I look for in order to know which one could I use instead?
Thanks in advantage.

**Comment by ndr-brt** [comment]
the module `transfer-data-plane` has been deprecated in favor of `transfer-data-plane-signaling`, all the control-plane/data-plane communication has changes since. Details: https://github.com/eclipse-edc/Connector/blob/main/docs/developer/data-plane-signaling/data-plane-signaling.md

**Reply by JorgeBarto** [reply]
Thanks ndr-brt!!
So, in order to be updated, we must do like this:
    //implementation(project(":extensions:control-plane:transfer:**transfer-data-plane**"))
    implementation(project(":extensions:control-plane:transfer:**transfer-data-plane-signaling**"))

**Reply by ndr-brt** [reply]
yes, changes must be done also in the `data-plane`, please take a look at the `transfer` samples: they are already using the `data-plane-signaling` feature (please note that in the samples the `data-plane` is embedded in the `control-plane`:
https://github.com/eclipse-edc/Samples/tree/main/transfer

**Reply by 3rim** [reply]
@ndr-brt Hello,
can you explain what you mean with 

> data-plane is embedded in the control-plane

and what changes need to be done in the ``data-plane``

**Reply by ndr-brt** [reply]
EDC is modular, `control-plane` and `data-plane` are two distinct components, but they can be deployed in the same runtime.
